### ISSUE TITLE
Storage missing parameters.json in build folder, in CI/CD

### ISSUE THREAD

**Issue by daniellorenzin** [aws-amplify/amplify-cli#8912]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### Amplify CLI Version

7.3.5

### What operating system are you using?

Mac

### Amplify Categories

storage

### Amplify Commands

push

### Describe the bug

Hi, I just upgraded to 7.3.4, and the console build process is now throwing errors like: 

Error: File at path: '/codebuild/output/src###/src/sandpit/amplify/backend/storage/excelUpload/build/parameters.json' 

I'm just using the `amplifyPush --simple` as the build backend command

✖ An error occurred when pushing the resources to the cloud
2021-11-17T03:00:43.611Z [WARNING]: ✖ There was an error initializing your environment.
2021-11-17T03:00:43.625Z [INFO]: [0mError: File at path: '/codebuild/output/src744360090/src/sandpit/amplify/backend/storage/excelUpload/build/parameters.json'

### Expected behavior

Should generate the files required to deploy.

### Reproduction steps

Upgrade to 7.3.5 and push to ci/cd which is setup for a new environment build.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by kaustavghosh06** [comment]
@daniellorenzin Few questions
1) Could you please provide us with the entire error log/stack-trac from the build logs?
2) Can you pull this environment locally and let us know all the files in the `amplify/backend/storage` directory?

**Comment by daniellorenzin** [comment]
In my repo there is only amplify/backend/storage/cli-inputs.json

Attached is the output from a new build, as well as my amplify.yaml.

I also have a cdk deployment as part of the amplify build process.
[log (12).txt](https://github.com/aws-amplify/amplify-cli/files/7551429/log.12.txt)

<img width="350" alt="Screen Shot 2021-11-17 at 2 11 49 pm" src="https://user-images.githubusercontent.com/7563399/142128444-a8c3da3e-1fd5-4d55-b138-244f7de3eb4b.png">

**Comment by kaustavghosh06** [comment]
Got it. And you encountered this error when creating a new branch from Amplify Console?

**Comment by daniellorenzin** [comment]
Yeah, we have a rule ```Branch autodetection - patterns feature/autobuild/**```
So I can spin up environments for testing and building

**Comment by kaustavghosh06** [comment]
@daniellorenzin Understood and was able to repro this scenario. If you re-deploy the same failed deployment, it should succeed - do give it a shot and let me know :)

**Comment by daniellorenzin** [comment]
Wasn't the case, because the last environment didn't successfully build it creates yet another new environment.
This might be an edge case but its when the backend name gets generated: ```Random environment name sherill generated```

The second time around I use the AWS_BACKEND_ENVIRONMENT_ARN to read out the name for the backend, and set it as the USER_BRANCH. Or else I get yet another environment with a concat name from the BRANCH name.

```if [ -z "$USER_BRANCH" ]; then USER_BRANCH=${AWS_BACKEND_ENVIRONMENT_ARN##*/}; fi;```

**Comment by kaustavghosh06** [comment]
Ah! Got it. I have pushed a fix for it  (#8913 ) and testing it right now before releasing.
